From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Mon, 9 Mar 2020 22:17:25 -0700
Subject: chore: allow disabling crypto tests

Upstreamed in https://github.com/nodejs/node/pull/31268. Allows for skipping
tests which require OpenSSL; Electron doesn't support all of OpenSSL and so need
to forcibly disable most crypto tests anyway. This makes that easier.

diff --git a/test/common/index.js b/test/common/index.js
index e93c73820afc11c982e42062731d2fdd1334e33e..f5387a426a24da759fc030ddca5e4e4fd2cc2d54 100644
--- a/test/common/index.js
+++ b/test/common/index.js
@@ -48,7 +48,8 @@ if (isMainThread)
 
 const noop = () => {};
 
-const hasCrypto = Boolean(process.versions.openssl);
+const hasCrypto = Boolean(process.versions.openssl) &&
+                  !process.env.NODE_SKIP_CRYPTO;
 
 // Check for flags. Skip this for workers (both, the `cluster` module and
 // `worker_threads`) and child processes.
